fix: form rerender on name change

zaCKoZAck0 opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->
1. Fixes the form preview re-render on name change.
1.1. This issue was caused by unnecessary Resetting question progress on name change.

Fixes # (issue)
Fixed: #968 

<!-- Please provide a screenshots or a loom video for visual changes to speed up reviews
 Loom Video: https://www.loom.com/
-->

## Type of change

<!-- Please mark the relevant points by using [x] -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- Test A
   - Try changing name while editing form, notice for re-render in the form preview. 
- Test B
  - Implement a method or change name as fast as possible to check if preview is still breaking. 

## Checklist

<!-- We're starting to get more and more contributions. Please help us making this efficient for all of us and go through this checklist. Please tick off what you did  -->

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

Dhruwang commented:
Hey @zaCKoZAck0 , thank you for the PR 😊
But unfortunately, this breaks our template page preview 

https://github.com/formbricks/formbricks/assets/67850763/f17cbd02-4466-483c-95a8-09dd343294dc

zaCKoZAck0 commented:
Hi, I've made the following changes
1. Removed dependency of resetting question progress from survey.name.
2. Rewritten the effect such that questions progress resetting is dependent on survey.questions.length, that way it only changes when number of question changes between two templates.

Dhruwang commented:
Hey @zaCKoZAck0 , thank you for the changes 😊
But I don't think checking based on survey question count is the right approach for this, because there are some templates which consist of same number of questions and that again breaks the progress bar 

https://github.com/formbricks/formbricks/assets/67850763/b1722a0d-2d25-49d1-bd93-0f40c0212e1e
